frontend/packages/integration-tests-cypress/tests/app/demo-dynamic-plugin.spec.ts

```diff
@@ -0,0 +1,184 @@
+      const yamlManifestUpdatedPullSpec = yamlManifest.map((yamlDefinition) => {
+        if (yamlDefinition.kind === 'Deployment' && !isLocalDevEnvironment() && PLUGIN_PULL_SPEC) {
+          return _.defaultsDeep(
+            {},
+            {
+              spec: {
+                template: {
+                  spec: {
+                    containers: [{ name: 'console-demo-plugin', image: PLUGIN_PULL_SPEC }],
+                  },
+                },
+              },
+            },
+            yamlDefinition,
+          );
+        }
+        return yamlDefinition;
+      });
+      const finalYAMLText = yamlManifestUpdatedPullSpec.map((obj) => safeDump(obj)).join('---\n');
+      yamlEditor.isImportLoaded();
+      yamlEditor.setEditorContent(finalYAMLText).then(() => {
+        yamlEditor.clickSaveCreateButton();
+        cy.get(errorMessage).should('not.exist');
+        cy.byTestID('success-icon').should('have.length', 5);
+        cy.byTestID('import-more-yaml').should('exist');
+        enableDemoPlugin(true);
+      });
+    });
+  });
+
+  after(() => {
+    enableDemoPlugin(false);
+    cy.deleteProject(PLUGIN_NAME);
+    cy.logout();
+  });
+
+  afterEach(() => {
+    checkErrors();
+  });
+
+  it(`test Dynamic Nav items`, () => {
+    const dynamicNavIDs = ['1', '2'];
+    dynamicNavIDs.forEach((id) => dynamicNavTest(id));
+  });
+
+  it(`test Test Consumer nav item`, () => {
+    nav.sidenav.clickNavLink(['Demo Plugin', 'Test Consumer']);
+    cy.get('.pf-c-title.pf-m-2xl').should('contain', 'Extensions of type Console.flag/Model');
+    cy.get('.pf-c-card').should('have.length', 20);
+  });
+
+  it(`test Test Utilities nav item`, () => {
+    nav.sidenav.clickNavLink(['Demo Plugin', 'Test Utilities']);
+    cy.get('.pf-c-title.pf-m-2xl').should('contain', 'Utilities from Dynamic Plugin SDK');
+    cy.get('.pf-c-card__title').should('contain', 'Utility: consoleFetchJSON');
+  });
+
+  it(`test List Page nav item`, () => {
+    const podName = 'openshift-state-metrics';
+    nav.sidenav.clickNavLink(['Demo Plugin', 'List Page']);
+    listPage.titleShouldHaveText('OpenShift Pods List Page');
+    listPage.rows.shouldBeLoaded();
+    listPage.filter.byName(podName);
+    listPage.rows.shouldExist(podName);
+  });
+
+  it(`test K8s API nav item`, () => {
+    nav.sidenav.clickNavLink(['Demo Plugin', 'K8s API']);
+    cy.get('.pf-c-title.pf-m-2xl').should('contain', 'K8s API from Dynamic Plugin SDK');
+    cy.get('.pf-c-button.pf-m-primary.pf-m-block').should('contain', 'k8sCreate');
+    cy.get('.pf-c-button.pf-m-primary.pf-m-block').should('contain', 'k8sGet');
+    cy.get('.pf-c-button.pf-m-primary.pf-m-block').should('contain', 'k8sPatch');
+    cy.get('.pf-c-button.pf-m-primary.pf-m-block').should('contain', 'k8sUpdate');
+    cy.get('.pf-c-button.pf-m-primary.pf-m-block').should('contain', 'k8sList');
+    cy.get('.pf-c-button.pf-m-primary.pf-m-block').should('contain', 'k8sDelete');
```

Comment:
Consider adding to `cy` commands,

```
Cypress.Commands.add('subNavShouldExist', (selector: string) =>
  cy.get('.pf-c-button.pf-m-primary.pf-m-block').should('contain', selector);
);

cy.subNavShouldExist(k8sCreate);
```

Reply:
@dtaylor113 used the existing byButtonText for these thanks for the tip. Also added a test id for the title.